doggo/src/main.rs

```diff
@@ -52,37 +52,54 @@ fn rate_pupper(mut conn: DoggersDb, rating: Form<Rating>) -> Result<&'static str
 }
 
 fn pupper_rating(mut conn:DoggersDb, name: &str) -> f64 {
-    for row in conn.0.query(format!("SELECT COALESCE(SUM(r.rating)/COUNT(r.rating),0.0) FROM doggers@replica.ratings as r WHERE r.pupper_name='{}'",name)).unwrap() {
-        let r: f64 = mysql::from_row(row.unwrap());
-        return r
-    }
-    return 0.0
+    return conn.0.query(format!("SELECT COALESCE(SUM(r.rating)/COUNT(r.rating),0.0) FROM doggers@replica.ratings as r WHERE r.pupper_name='{}'",name)).ok().and_then(|mut result| {
```

Comment:
Now that vitess supports prepared statements, we can change this to use the prep_exec method:
```suggestion
    return conn.0.prep_exec(
        "SELECT COALESCE(SUM(r.rating)/COUNT(r.rating),0.0) FROM doggers@replica.ratings as r WHERE r.pupper_name=?"), 
        (name,)
    ).ok().and_then(|mut result| {
```

Reply:
Prepared statements are bad and we shouldn't use them, even though vitess supports them they are strictly worse then just sending in the query